src/crawlee/http_crawler/_http_parser.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Generic
+
+from typing_extensions import override
+
+from crawlee.http_crawler._http_crawling_context import TParseResult
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from crawlee.http_clients import HttpResponse
+
+
+@dataclass(frozen=True)
+class BlockedInfo:
+    """Information about whether the crawling is blocked. If reason is empty, then it means it is not blocked."""
+
+    reason: str
+
+    def __bool__(self) -> bool:
+        """No reason means no blocking."""
+        return bool(self.reason)
+
+
+class StaticContentParser(Generic[TParseResult], ABC):
```

Comment:
@Pijukatel what do you think about this one?

Reply:
I didn't get it :-) I am just guessing. Did you mean something like this? 

class Result[TParsedResult]:
  value=...
  match_selector=...
  some_other_stuff = ...

Reply:
It makes sense from the point of view that extending it will be very easy. On the other hand it will be one extra layer of access to get the stuff. Also once we decide to add more stuff, we can add it in the context and not in this class anyway. So maybe I would just keep parsed_result for just return value of parser (which can be class in some custom crawler. So I think current PR offers it, but it does not enforce it.) 
I am usually very careful about trying to predict future. It is too hard to reason about potential future stuff.

Reply:
Yeah, most of my reasons to add that are contrived at best and invalid at worst. Let's keep it this way.